storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestConstraintSolverStrategy.java

```diff
@@ -239,42 +435,50 @@ public void testScheduleLargeExecutorConstraintCount(int parallelismMultiplier)
         scheduler.schedule(topologies, cluster);
 
         boolean scheduleSuccess = isStatusSuccess(cluster.getStatus(topo.getId()));
-        LOG.info("testScheduleLargeExecutorCount scheduling {} with {}x executor multiplier", scheduleSuccess ? "succeeds" : "fails",
-                parallelismMultiplier);
+        LOG.info("testScheduleLargeExecutorCount scheduling {} with {}x executor multiplier, consolidatedConfigFlag={}",
+                scheduleSuccess ? "succeeds" : "fails", parallelismMultiplier, consolidatedConfigFlag);
         Assert.assertTrue(scheduleSuccess);
     }
 
     @Test
     public void testIntegrationWithRAS() {
-        List<List<String>> constraints = new LinkedList<>();
-        addContraints("spout-0", "bolt-0", constraints);
-        addContraints("bolt-1", "bolt-1", constraints);
-        addContraints("bolt-1", "bolt-2", constraints);
-        List<String> spread = new LinkedList<>();
-        spread.add("spout-0");
+        if (!consolidatedConfigFlag) {
+            LOG.info("Skipping test since bolt-1 maxCoLocationCnt=10 requires consolidatedConfigFlag=true, current={}", consolidatedConfigFlag);
+            return;
+        }
 
         Map<String, Object> config = Utils.readDefaultConfig();
         config.put(Config.TOPOLOGY_SCHEDULER_STRATEGY, ConstraintSolverStrategy.class.getName());
-        config.put(Config.TOPOLOGY_SPREAD_COMPONENTS, spread);
-        config.put(Config.TOPOLOGY_RAS_CONSTRAINTS, constraints);
         config.put(Config.TOPOLOGY_RAS_CONSTRAINT_MAX_STATE_SEARCH, MAX_TRAVERSAL_DEPTH);
         config.put(Config.TOPOLOGY_WORKER_MAX_HEAP_SIZE_MB, 100_000);
         config.put(Config.TOPOLOGY_PRIORITY, 1);
         config.put(Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT, 10);
         config.put(Config.TOPOLOGY_COMPONENT_RESOURCES_ONHEAP_MEMORY_MB, 100);
         config.put(Config.TOPOLOGY_COMPONENT_RESOURCES_OFFHEAP_MEMORY_MB, 0.0);
 
+        List<List<String>> constraints = new LinkedList<>();
+        addConstraints("spout-0", "bolt-0", constraints);
+        // commented out unsatisfiable constraint since there are 300 executors and cannot fit on 30 nodes, added as spread
+        // addConstraints("bolt-1", "bolt-1", constraints);
```

Comment:
Remove this if it is not needed?

Reply:
Added back. Constraint is satifisfiable when bolt-1 -> bolt-1 is not interpreted as maxNodeCoLocationCnt (fixed in ConstraintSolverStrategy)